src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than zero');
+    }
+
+    const sort = [a, b, c].sort((d, f) => d - f);
+
+    if (sort[2] >= sort[0] + sort[1]) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
+    }
+
+    this.color = color;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
 
-export class Circle implements Figure {}
+  color: 'red' | 'green' | 'blue';
 
-export class Rectangle implements Figure {}
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
The parameter `radius` should be prefixed with `this.` to refer to the instance variable. The current code will throw an error because `radius` is not defined in this scope.